src/coreComponents/physicsSolvers/multiphysics/SinglePhasePoroelasticKernel.hpp

```diff
@@ -0,0 +1,447 @@
+                                                                                stack.localFlowDofIndex,
+                                                                                stack.localDispFlowJacobian[numDofPerTestSupportPoint * localNode + dim],
+                                                                                nPDof );
+
+      }
+    }
+
+    for( localIndex i = 0; i < nPDof; ++i )
+    {
+      localIndex const dof = LvArray::integerConversion< localIndex >( stack.localFlowDofIndex[ i ] - m_dofRankOffset );
+      if( dof < 0 || dof >= m_matrix.numRows() )
+        continue;
+      m_matrix.template addToRowBinarySearchUnsorted< parallelDeviceAtomic >( dof,
+                                                                              stack.localRowDofIndex,
+                                                                              stack.localFlowDispJacobian[i],
+                                                                              nUDof );
+      m_matrix.template addToRowBinarySearchUnsorted< parallelDeviceAtomic >( dof,
+                                                                              stack.localFlowDofIndex,
+                                                                              stack.localFlowFlowJacobian[i],
+                                                                              nPDof );
+      RAJA::atomicAdd< parallelDeviceAtomic >( &m_rhs[ dof ], stack.localFlowResidual[i] );
+    }
+
+
+
+    return maxForce;
+  }
+
+
+
+protected:
+  /// The array containing the nodal position array.
+  arrayView2d< real64 const, nodes::REFERENCE_POSITION_USD > const m_X;
+
+  /// The rank-global displacement array.
+  arrayView2d< real64 const, nodes::TOTAL_DISPLACEMENT_USD > const m_disp;
+
+  /// The rank-global incremental displacement array.
+  arrayView2d< real64 const, nodes::INCR_DISPLACEMENT_USD > const m_uhat;
+
+  /// The gravity vector.
+  real64 const m_gravityVector[3];
+  real64 const m_gravityAcceleration;
+
+  /// The rank global densities
+  arrayView2d< real64 const > const m_solidDensity;
+  arrayView2d< real64 const > const m_fluidDensity;
+  arrayView1d< real64 const > const m_fluidDensityOld;
+  arrayView2d< real64 const > const m_dFluidDensity_dPressure;
+
+  /// The global degree of freedom number
+  arrayView1d< globalIndex const > const m_flowDofNumber;
+
+  /// The rank-global fluid pressure array.
+  arrayView1d< real64 const > const m_fluidPressure;
+
+  /// The rank-global delta-fluid pressure array.
+  arrayView1d< real64 const > const m_deltaFluidPressure;
+
+  /// The rank-global reference porosity array
+  arrayView1d< real64 const > const m_poroRef;
+
+  arrayView1d< real64 const > const m_bulkModulus;
+
+  real64 const m_biotCoefficient;
+
+
+};
+
+
+} // namespace SolidMechanicsLagrangianFEMKernels
+
+} // namespace geosx
+
+#include "finiteElement/kernelInterface/SparsityKernelBase.hpp"
```

Comment:
Can this include be moved to the top of the file?

Reply:
The include location mimics what done in SolidMechanicsSmallStrainQuasiStaticKernel.hpp (see [here](https://github.com/GEOSX/GEOSX/blob/80ab48fb27b53db20977903ea4743d31dcd0d501/src/coreComponents/physicsSolvers/solidMechanics/SolidMechanicsSmallStrainQuasiStaticKernel.hpp#L334)).

Reply:
The one in SolidMechanics... seem to be the only one include that is placed at the bottom of a file. Any special reason for this?